### ISSUE TITLE
[Bug]: Always snap to grid not working since 1.4.3

### ISSUE THREAD

**Issue by TheToxin-git** [Comfy-Org/ComfyUI_frontend#1601]
### Frontend Version

1.4.3
![1732004222_432266961](https://github.com/user-attachments/assets/6b5f4458-0e79-4a3e-85a1-7b1071610a7c)

### Expected Behavior

Groups and Nodes should snap to grid, when Always snap to grid is on

### Actual Behavior

Groups and Nodes do not snap to grid, until shift-key is pressed, yet Always snap to grid is on (for demonstration purposes snap to grid size was set to 64):

https://github.com/user-attachments/assets/bf387b0a-3b8e-4306-81d3-cf0d34299cf7

### Steps to Reproduce

Update to 1.4.3 and turn on Always snap to grid? idk
![1732004284_824090828](https://github.com/user-attachments/assets/efc6d5da-f414-4e59-9d41-bc55cce4026d)

### Debug Logs

```powershell
Total VRAM 11934 MB, total RAM 31887 MB
pytorch version: 2.5.1+cu124
xformers version: 0.0.28.post3
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3060 : cudaMallocAsync
Using xformers cross attention
Initializing frontend: Comfy-Org/ComfyUI_frontend@latest, requesting version details from GitHub...
[Prompt Server] web root: /home/thetoxin/root/downloads/github/ComfyUI/web_custom_versions/Comfy-Org_ComfyUI_frontend/1.4.3
Skipping loading of custom nodes
Starting server

To see the GUI go to: http://127.0.0.1:8188
```

### Browser Logs

```
Running on 127.0.0.1:8188 api.ts:56:12
ComfyUI Front-end version: 1.4.3 logging.ts:333:13
unreachable code after return statement three.module.js:8173:11
load and start tracking of workflow workflows/Unsaved Workflow.json workflowStore.ts:73:12
[workflowStore] open workflow workflows/Unsaved Workflow.json workflowStore.ts:191:12
```

### Setting JSON

[comfy.settings.json](https://github.com/user-attachments/files/17811799/comfy.settings.json)

### What browsers do you use to access the UI ?

Mozilla Firefox

**Comment by JorgeR81** [comment]
> Groups and Nodes do not snap to grid, until shift-key is pressed.

Yeah, I agree. I always want to snap to grid,
( on Windows 10, pressing `Shift` to many times, enables the "sticky keys" feature ).

Also, groups and nodes don't snap to grid when we resize them.
Not even if we press the `Shift` key, while resizing.

**Comment by LukeG89** [comment]
@huchenlei @webfiltered Snap to grid still doesn't work in 1.4.4, you still need to press Shift (tested both on Chrome and Edge)


https://github.com/user-attachments/assets/9400496f-da7f-4504-9c41-7159db2ae8fd


### Debug logs:
```powershell
E:\>.\python_embeded\python.exe -s ComfyUI\main.py --windows-standalone-build --disable-all-custom-nodes --preview-size 1024 --front-end-version Comfy-Org/ComfyUI_frontend@latest
Total VRAM 8192 MB, total RAM 65277 MB
pytorch version: 2.6.0.dev20241115+cu124
Set vram state to: NORMAL_VRAM
Device: cuda:0 NVIDIA GeForce RTX 3070 Ti Laptop GPU : cudaMallocAsync
Using pytorch cross attention
Initializing frontend: Comfy-Org/ComfyUI_frontend@latest, requesting version details from GitHub...
[Prompt Server] web root: E:\ComfyUI\web_custom_versions\Comfy-Org_ComfyUI_frontend\1.4.4
Skipping loading of custom nodes
Starting server

To see the GUI go to: http://127.0.0.1:8188
```

### Console logs:
```console
Running on 127.0.0.1:8188
logging.ts:333 ComfyUI Front-end version: 1.4.4
Uncaught Error: NOT_REACHED
    at unsetKeybinding (keybindingStore.ts:234:11)
    at Proxy.loadUserKeybindings (keybindingStore.ts:259:23)
    at Proxy.wrappedAction (pinia.mjs:1405:26)
    at requestIdleCallback.timeout (GraphView.vue:152:28)
```

**Comment by TheToxin-git** [comment]
@LukeG89 Mine starts working after I re-enabled (Enabled -> Disabled -> Close Settings -> Open Settings -> Enabled) Always snap to grid option once. Try it. Maybe yours will work too.

On the console logs I have related issue, yet I do not formalized it properly. There is something wrong with keybindings migration to 1.4.3+. Keybindings are being reset after update and I believe that console error is related to that.

**Comment by LukeG89** [comment]
> @LukeG89 Mine starts working after I re-enabled (Enabled -> Disabled -> Close Settings -> Open Settings -> Enabled) Always snap to grid option once. Try it. Maybe yours will work too.

Yes, now it's working! 🎉 
Thanks for the tip! 

> On the console logs I have related issue, yet I do not formalized it properly yet. There is something wrong with keybindings migration to 1.4.3+. Keybindings are being reset after update and I believe that console error is related to that.

Yes, all my keybindings are gone too...

**Comment by LukeG89** [comment]
@webfiltered A different issue related to Snap to grid PR.

Apparently it broke guidelines from [ComfyUI-Custom-Scripts](https://github.com/pythongosssss/ComfyUI-Custom-Scripts)


https://github.com/user-attachments/assets/78268041-59d0-4701-bd8a-dfa126e43e4d

**Comment by LukeG89** [comment]
> On the console logs I have related issue, yet I do not formalized it properly. There is something wrong with keybindings migration to 1.4.3+. Keybindings are being reset after update and I believe that console error is related to that.

@TheToxin-git I've reported the bug: https://github.com/Comfy-Org/ComfyUI_frontend/issues/1612

**Comment by JorgeR81** [comment]
> Mine starts working after I re-enabled (Enabled -> Disabled -> Close Settings -> Open Settings -> Enabled) Always snap to grid option once. Try it. Maybe yours will work too.

This also worked for me, but we still don't have snap to grid when resizing groups or nodes.

**Comment by JorgeR81** [comment]
> Mine starts working after I re-enabled (Enabled -> Disabled -> Close Settings -> Open Settings -> Enabled) Always snap to grid option once. Try it. Maybe yours will work too.

When I restart ComfyUI, it stops working again.
So I have to repeat the process ( disable and reenable again ).

**Comment by LukeG89** [comment]
> When I restart ComfyUI, it stops working again. So I have to repeat the process ( disable and reenable again ).

@JorgeR81 Try disable it -> Refresh the page -> Enable it again

**Comment by JorgeR81** [comment]
> When I restart ComfyUI, it stops working again. So I have to repeat the process ( disable and reenable again ).
> 
> @JorgeR81 Try disable it -> Refresh the page -> Enable it again

Ah, yes. It worked ! Thanks.

Now we only need snapping while resizing. 

Is that supposed to be fixed here, or should I open a different issue ?

**Comment by webfiltered** [comment]
@JorgeR81 I'll just open one - I had planned to implement this properly at some point.

**Comment by LukeG89** [comment]
> > @JorgeR81 Try disable it -> Refresh the page -> Enable it again
> 
> Ah, yes. It worked ! Thanks.

I deleted my comment because it only works in the current workflow, if you open a new one snap to grid doesn't work

**Comment by LukeG89** [comment]
@webfiltered Why is it only working in the first workflow?


https://github.com/user-attachments/assets/fad92555-7ce5-49fe-9aed-d99d50a5cb44

**Comment by webfiltered** [comment]
It's a per-workflow setting - I made a mistake on a last-minute alteration.  Fixing it so it behaves as expected.

**Comment by alessandroperilli** [comment]
FYI, the fix was automatically enforced in v1.4.5. But when I upgraded to v1.4.6, I had to switch the toggle off and on again.
Without doing that, the Always Snap to Grid functionality seems broken again.

----
While we are at this, I have a question:

Both `AlwaysSnaptoGrid` and `SHIFT+Drag` to manually snap work fine with both nodes and groups. Thank you so much for that. 

But the functionally doesn't seem to work when you try to resize nodes and groups (I tried with both Vivaldi and Safari on macOS). Is this by design, in the sense that the feature was never implemented? Or is it a bug?

If the former, I'd be happy to open a dedicated feature request. Thank you!

**Comment by webfiltered** [comment]
Appreciate the info!  I've just submitted a fix to resolve the underlying issue with snap to grid (movement).

Resize is up next.  I was going to get some more framework stuff done first, but the community has really let the priorities be known on this one!  😆
